### PR TITLE
[RQ-58] Go tests for usage attribution endpoints

### PR DESCRIPTION
**Pull request by anthonypalmeira**
### What does this PR do?

Adds Golang tests for the following endpoints:

**Endpoint 1: Daily Custom reporting available files
Endpoint 2: Daily Custom reporting file URL
Endpoint 3: Monthly Custom reporting available files
Endpoint 4: Monthly Custom reporting file URL**

### Review checklist
Please check relevant items below:
- [ ] This PR includes all [newly recorded cassettes](/DEVELOPMENT.md) for any modified tests.

- [ ] This PR does not rely on API client schema changes.
    - [ ] The CI should be fully passing.
- [ ] Or, this PR relies on API schema changes and this is a Draft PR to include tests for that new functionality.
  - Note: CI shouldn't be run on this Draft PR, as its expected to fail without the corresponding schema changes.

### CODE REVIEW

File: tests/api/v1/datadog/api_usage_metering_test.go [hunk 1]
```diff
@@ -317,6 +317,10 @@ func TestUsageBillableSummary(t *testing.T) {
 }
 
 func TestSpecifiedDailyCustomReports(t *testing.T) {
+	if tests.GetRecording() == tests.ModeReplaying {
+		t.Skip("No reports are available yet for Org 321813")
+	}
```
Comment (jirikuncar) [inline tests/api/v1/datadog/api_usage_metering_test.go:322]: can we move it inside error handling?

```go
if err != nil {
	if tests.GetRecording() != tests.ModeReplaying && httpresp.StatusCode == 404 {
		t.Skip("No reports are available yet")
	} else {
		t.Errorf("Error getting ...")
	}
}
```

Comment (jirikuncar) [reply tests/api/v1/datadog/api_usage_metering_test.go:322]: These tests are scheduled to run every 3 hours so eventually we should see a successful trace.

Comment (anthonypalmeira) [reply tests/api/v1/datadog/api_usage_metering_test.go]: Done

File: tests/api/v1/datadog/api_usage_metering_test.go [hunk 2]
```diff
@@ -316,6 +316,86 @@ func TestUsageBillableSummary(t *testing.T) {
 	assert.Equal(float64(0.9), usageKeys.GetPercentageInAccount())
 }
 
+func TestSpecifiedDailyCustomReports(t *testing.T) {
+	ctx, finish := WithRecorder(WithTestAuth(context.Background()), t)
+	defer finish()
+	assert := tests.Assert(ctx, t)
+	currentDate := time.Now().AddDate(0, 0, -1)  // only have reports from day before
+	reportID := currentDate.Format("2006-01-02") // this is only a date format example YYYY-MM-DD
```
Comment (jirikuncar) [inline tests/api/v1/datadog/api_usage_metering_test.go:324]: 👏 nice

**Pull request closed without merging**